File: src/main/java/com/thatameliah/SmartCardRegister/Handlers/FileHandler.kt
package com.thatameliah.SmartCardRegister.Handlers

import java.io.File
import java.io.IOException
import java.nio.file.Files
import java.nio.file.Path
import java.nio.file.Paths

object FileHandler {
    val savesDirectory: Path = Paths.get("saves")

    // Gets the Path to the specified file
    @JvmStatic
    private fun getPathToFIle(filename: String): Path {
        return savesDirectory.resolve(filename)
    }

    /**
     * Ensures the saves directory exists
     */
    @JvmStatic
    @Throws(IOException::class)
    private fun ensureDirectory() {
        if (!Files.exists(savesDirectory)) {
            Files.createDirectories(savesDirectory)
        }
    }

    /**
     * Writes a string to a file
     *
     * @param content The String to be written
     * @param file The file to write to
     */
    @JvmStatic
    fun writeToFile(content: String, file: File) {
        val path = file.toPath()
        try {
            Files.createDirectories(path.getParent())
            Files.writeString(path, content)
            println("Successfully wrote to $path")
        } catch (e: IOException) {
            System.err.println("Error writing to file " + path + ": " + e.message)
        }
    }

    /**
     * Reads the contents of a file
     *
     * @param file The file to read from
     * @return The string contents of the file, or null if it is empty
     */
    @JvmStatic
    fun readFile(file: File): String? {
        val path = file.toPath()
        if (!Files.exists(path)) {
            System.err.println("File not found: $path")
            return null
        }
        try {
            return Files.readString(path)
        } catch (e: IOException) {
            System.err.println("Error reading file " + path + ": " + e.message)
            return null
        }
    }
}

File: src/main/java/com/thatameliah/SmartCardRegister/Handlers/FileHandler.scala
package com.thatameliah.SmartCardRegister.Handlers

import java.io.{File, IOException}
import java.nio.file.{Files, Path, Paths}

object FileHandler {
  private val SAVES_DIRECTORY: Path = Paths.get("saves")

  private def GetPathToFile(filename: String) = SAVES_DIRECTORY.resolve(filename)

  @throws[IOException]
  private def EnsureDirectory(): Unit = {
    if (!Files.exists(SAVES_DIRECTORY)) Files.createDirectories(SAVES_DIRECTORY)
  }

  def WriteToFile(content: String, file: File): Unit = {
    val path: Path = file.toPath
    try {
      Files.createDirectories(path.getParent)
      Files.writeString(path, content)
      print("Successfully wrote to " + path)
    } catch {
      case err: IOException => print("Error writing to file " + path + ": " + err.getMessage)
    }
  }

  def ReadFile(file: File): String = {
    val path = file.toPath
    if (!Files.exists(path)) {
      System.err.println("File not found: " + path)
      return null
    }
    try Files.readString(path)
    catch {
      case e: IOException =>
        System.err.println("Error reading file " + path + ": " + e.getMessage)
        null
    }
  }

  // Delete file if exists
  def DeleteFile(filename: String): Unit = {
    val path = GetPathToFile(filename)
    try if (Files.deleteIfExists(path)) System.out.println("Deleted file " + path)
    catch {
      case e: IOException =>
        System.err.println("Failed to delete file " + path + ": " + e.getMessage)
    }
  }
}
